### PR TITLE
Add CC13x4 Example Application support

### PR DESCRIPTION
**Pull request by adabreuti**
Add initial CC13x4 Application support
Upgrade lock-app to utilize latest UART driver instances
Update README's with clearer CHIP CLI instructions

### CODE REVIEW

**Review by ReneJosefsen — APPROVED** [review]

**Review by mgarb1 — APPROVED** [review]

**Comment by adabreuti** [comment]
@bzbarsky-apple I believe the only CI check that's failing is optional, if that's not the case let me know and i'll investigate further. From my understanding, however, the example application changes should not have affected the Darwin test suite.

**Comment by bzbarsky-apple** [comment]
@adabreuti The darwin failure is definitely a random failure; I retriggered that job.  But this also needs review from some non-Apple reviewer (PRs need two reviews from different companies).

File: integrations/docker/images/chip-build-ti/Dockerfile
Comment (adabreuti) [inline integrations/docker/images/chip-build-ti/Dockerfile:1]: @bzbarsky-apple Do you know of the contact for pushing the updates for the TI Docker image? The current TI CI will fail without the latest versions described here.

Comment (bzbarsky-apple) [reply integrations/docker/images/chip-build-ti/Dockerfile:1]: @adabreuti The right way to handle things involving Docket updates is:

1. A PR to update the integrations/ bits to create the new docker images.  Get that reviewed and landed.  Don't forget to change `integrations/docker/images/chip-build/version` for the version update as part of this PR; I just realized this PR is missing this very important bit.
2. Wait for docker hub to be updated.  Ping @andy31415 and @woody-apple as needed to make sure this happens.
3. A second PR to actually use the new images, including changing the image version used in the workflow file to match the one from the first PR.

Comment (adabreuti) [reply integrations/docker/images/chip-build-ti/Dockerfile:1]: Thanks for the feedback @bzbarsky-apple, I've created the following PR based on your suggestions and have removed the changes from this one.
https://github.com/project-chip/connectedhomeip/pull/26730

**Review by andy31415 — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@adabreuti For future reference: the requirement is that the merge to master not fail.  Being up-to-date is not viable, as you note; we have too many commits happening for that.

**Pull request merged**